iothub/service/src/Jobs/ScheduledJobsClient.cs

```diff
@@ -0,0 +1,386 @@
+                {
+                    JobId = jobId,
+                    JobType = JobType.ScheduleDeviceMethod,
+                    CloudToDeviceMethod = cloudToDeviceMethod,
+                    QueryCondition = queryCondition,
+                    StartTime = startTimeUtc,
+                    MaxExecutionTimeInSeconds = maxExecutionTimeInSeconds
+                };
+
+                return await CreateAsync(jobRequest, cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                if (Logging.IsEnabled)
+                    Logging.Error(this, $"{nameof(ScheduleDeviceMethodAsync)} threw an exception: {ex}", nameof(ScheduleDeviceMethodAsync));
+                throw;
+            }
+            finally
+            {
+                if (Logging.IsEnabled)
+                    Logging.Enter(this, $"jobId=[{jobId}], queryCondition=[{queryCondition}]", nameof(ScheduleDeviceMethodAsync));
+            }
+        }
+
+        /// <summary>
+        /// Creates a new Job to update twin tags and desired properties on one or multiple devices
+        /// </summary>
+        /// <param name="jobId">Unique Job Id for this job</param>
+        /// <param name="queryCondition">Query condition to evaluate which devices to run the job on</param>
+        /// <param name="twin">Twin object to use for the update</param>
+        /// <param name="startTimeUtc">Date time in Utc to start the job</param>
+        /// <param name="maxExecutionTimeInSeconds">Max execution time in seconds, i.e., ttl duration the job can run</param>
+        /// <param name="cancellationToken">Task cancellation token</param>
+        /// <returns>A JobResponse object</returns>\
+        /// <exception cref="ArgumentNullException">Thrown when the provided jobId or queryCondition or twin or startTimeUtc or maxExecutionTimeInSeconds is null.</exception>
+        /// <exception cref="ArgumentException">Thrown if the jobId or queryCondition is empty or whitespace.</exception>
+        /// <exception cref="IotHubException">
+        /// Thrown if IoT hub responded to the request with a non-successful status code. For example, if the provided
+        /// request was throttled, <see cref="IotHubThrottledException"/> is thrown. For a complete list of possible
+        /// error cases, see <see cref="Common.Exceptions"/>.
+        /// </exception>
+        /// <exception cref="HttpRequestException">
+        /// If the HTTP request fails due to an underlying issue such as network connectivity, DNS failure, or server
+        /// certificate validation.
+        /// </exception>
+        /// <exception cref="OperationCanceledException">If the provided cancellation token has requested cancellation.</exception>
+        public virtual Task<JobResponse> ScheduleTwinUpdateAsync(string jobId, string queryCondition, Twin twin, DateTime startTimeUtc, long maxExecutionTimeInSeconds, CancellationToken cancellationToken = default)
```

Comment:
This API is pretty heavy on parameters. I'd consider making a concrete type to encapsulate a few of these parameters

Reply:
can you give an example?

Reply:
Something like:

```csharp
public virtual Task<JobResponse> ScheduleTwinUpdateAsync(string jobId, ScheduledTwinUpdate scheduledTwinUpdate, CancellationToken cancellationToken = default);

public class ScheduledTwinUpdate
{
    public string queryCondition { get; set; }

    public Twin twin { get; set; }

    public DateTime startTimeUtc { get; set; }

    public long? maxExecutionTimeInSeconds { get; set; }
}
```

Doing it like this allows us to provide defaults values, leave values null that are optional, etc.

Reply:
The jobId parameter is interesting because I think it is required, but can be an arbitrary value. We may be able to label it as an optional parameter and provide it for the user if they don't specify a value.

Reply:
To build on this a bit, the usual pattern we follow for APIs like these is

```csharp
public void SomeMethod(string requiredParam1, string requiredParam2, SomeMethodRequestOptions options = default, CancellationToken cancellationToken = default)
```

Where ```SomeMethodRequestOptions``` contains all the optional parameters for the request. An example of this style can be seen [here](https://github.com/Azure/azure-iot-sdk-csharp/blob/b7e0cb9a4b94dff1a4ed009eeb1ee0ac61c6bb15/iothub/service/src/DigitalTwin/DigitalTwinsClient.cs#L125).

In order to make this API both concise and flexible, you should aim to follow this pattern. In order to do that, you'll need to figure out which of these parameters are required by the service and which are optional.